syntax = "proto3";

// ######################
// ###### REQUESTS ###### 
// ######################

message RecommendedGamesRequest {
    int32 max_results = 1;
}

// #######################
// ###### RESPONSES ###### 
// #######################

message GameInfo {
	string id = 1;
	string name = 2;
    int32 reviews_number = 3;
    int32 recommend_count = 4;
}

message GamesInfoResponse {
    repeated GameInfo games = 1;
}

// ######################
// ###### SERVICES ###### 
// ######################

service Steam {
	rpc RecommendedGames (RecommendedGamesRequest) returns (GamesInfoResponse);
}